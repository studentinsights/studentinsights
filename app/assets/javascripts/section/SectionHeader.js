import React from 'react';
const Routes = window.shared.Routes;

class SectionHeader extends React.Component {

  handleChangeSection(event) {
    window.location.href = Routes.section(event.target.value);
  }

  render () {
    const section = this.props.section;

    return (
      <div className='SectionHeader'>
        <div id="section-header-info">
          <h1>{section.section_number}</h1>
          <p id="course-info">{section.course_description} {'(' + section.course_number + ')'}</p>
          <p id="section-detail">Room {section.room_number} {this.renderBulletSpacer()} Schedule: {section.schedule} {this.renderBulletSpacer()} Term: {section.term_local_id}</p>
        </div>
        <div><p>Section: {this.renderSectionSelector()}</p></div>
      </div>
    );
  }

  renderSectionSelector() {
    return (
      <span>
        <select id="section-select" value={this.props.section.id} onChange={this.handleChangeSection}>
            {this.props.sections.map(section => {
              return (
                <option key={section.id} value={section.id}>
                  {section.section_number}
                </option>
              );
            },this)}
        </select>
      </span>
    );
  }

  renderBulletSpacer() {
    return (
      <span>
        â€¢
      </span>
    );
  }
}

SectionHeader.propTypes = {
  section: React.PropTypes.object.isRequired,
  sections: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
  educators: React.PropTypes.arrayOf(React.PropTypes.object).isRequired
};

export default SectionHeader;
