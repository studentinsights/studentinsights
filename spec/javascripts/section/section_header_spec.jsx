import renderer from 'react-test-renderer';
import SectionHeader from '../../../app/assets/javascripts/section/section_header.jsx';

describe('SectionHeader', function() {
  
  it('renders correctly', () => {
    const props = {
      educators: [
        { }
      ],
      sections: [
        { id: 1, section_number: 'Art-1'},
        { id: 2, section_number: 'Art-2'},
        { id: 3, section_number: 'Art-3'}
      ],
      section: { id: 1, section_number: 'Art-1', course_number: 'Art', course_description: 'Awesome Art Class', term_local_id: '9', schedule: '3(M-R)', room_number: '304' },
    };
    
    const tree = renderer.create(
      <SectionHeader {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/*

propTypes: {
    section: React.PropTypes.object.isRequired,
    sections: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
    educators: React.PropTypes.arrayOf(React.PropTypes.object).isRequired
  },

*/