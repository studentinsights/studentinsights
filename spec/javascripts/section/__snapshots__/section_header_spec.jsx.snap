// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader renders correctly 1`] = `
<div
  className="SectionHeader"
>
  <div
    id="section-header-info"
  >
    <h1>
      Art-1
    </h1>
    <p
      id="course-info"
    >
      Awesome Art Class
       
      (Art)
    </p>
    <p
      id="section-detail"
    >
      Room 
      304
       
      <span>
        •
      </span>
       Schedule: 
      3(M-R)
       
      <span>
        •
      </span>
       Term: 
      9
    </p>
  </div>
  <div>
    <p>
      Section: 
      <span>
        <select
          id="section-select"
          onChange={[Function]}
          value={1}
        >
          <option
            value={1}
          >
            Art-1
          </option>
          <option
            value={2}
          >
            Art-2
          </option>
          <option
            value={3}
          >
            Art-3
          </option>
        </select>
      </span>
    </p>
  </div>
</div>
`;
